### PR TITLE
Mid round rulesets refactor

### PR DESCRIPTION
**Pull request by ShiftyRail**
- No longer the same code with slight variations for Nukies, Wiz, and the like.
- Grinch (but he's dead now) will no longer spawn mindless.
- Nukies can actually fire midround. The role_id wasn't the good one.
- No longer uses a key as reference and searches a list, but rather a direct reference to the mob.

All tested.

:cl:
- bugfix: Changes to midround rulesets in dynamic mode, notably, making Nuke Ops able to fire again.

### CODE REVIEW

File: code/datums/gamemode/dynamic/dynamic_rulesets_midround.dm [hunk 1]
```diff
@@ -41,7 +97,6 @@
 /datum/dynamic_ruleset/midround/autotraitor/execute()
 	var/mob/M = pick(living_players)
 	assigned += M
-	living_players -= M
```
Comment (ShiftyRail) [inline code/datums/gamemode/dynamic/dynamic_rulesets_midround.dm]: Unintended

File: code/modules/mob/mob.dm [hunk 2]
```diff
@@ -108,6 +108,10 @@
 			qdel(A)
 		orient_object = null
 
+	if (ticker && istype(ticker.mode, /datum/gamemode/dynamic))
```
Comment (DamianX) [inline code/modules/mob/mob.dm]: shitcode
call a proc defined on mode and overridden in dynamic instead

File: code/datums/gamemode/dynamic/dynamic_rulesets.dm [hunk 3]
```diff
@@ -115,16 +115,16 @@
 	if (!searching)
 		return
 	if(jobban_isbanned(M, role_category) || isantagbanned(M))
-		to_chat(M, "<span class='danger'>Banned from [role_category].</span>")
+		to_chat(M, "<span class='danger'>Banned from [initial(role_category.id)].</span>")
 		to_chat(M, "<span class='warning'>Your application has been discarded due to past conduct..</span>")
 		return
 	if(M.key in applicants)
```
Comment (ShiftyRail) [inline code/datums/gamemode/dynamic/dynamic_rulesets.dm]: This should be, `M in applicants`.

**Comment by ShiftyRail** [comment]
It's a purely back-end change ; does it needs a CL ?

**Comment by ShiftyRail** [comment]
Consider merging this because it fixes nuke ops

File: code/datums/gamemode/dynamic/dynamic_rulesets_midround.dm [hunk 4]
```diff
@@ -141,69 +158,34 @@
...
-/datum/dynamic_ruleset/midround/nuclear/acceptable(var/population=0,var/threat=0)
+/datum/dynamic_ruleset/midround/from_ghosts/faction_based/nuclear/acceptable(var/population=0,var/threat=0)
 	if (locate(/datum/dynamic_ruleset/roundstart/nuclear) in mode.executed_rules)
 		return 0//unavailable if nuke ops were already sent at roundstart
 	return ..()
 
-/datum/dynamic_ruleset/midround/nuclear/ready(var/forced = 0)
+/datum/dynamic_ruleset/midround/from_ghosts/faction_based/nuclear/ready(var/forced = 0)
 	if (required_candidates > (dead_players.len + list_observers.len))
+		to_chat(world, "[required_candidates] > [dead_players.len] + [list_observers.len]")
```
Comment (ShiftyRail) [inline code/datums/gamemode/dynamic/dynamic_rulesets_midround.dm:168]: lmao

**Pull request merged**